src/test/lib/ApiTest.cpp

```diff
@@ -1152,6 +1238,77 @@ DummyStreamCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+struct CloseFromCallbackContext {
+    uint16_t CloseCount;
+    uint16_t CurrentCount;
+
+    static QUIC_STATUS StreamCallback(_In_ MsQuicStream*, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT*) {
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS Callback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event, _In_ bool IsServer) {
+        QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+        auto Ctx = (CloseFromCallbackContext*)Context;
+
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, StreamCallback, &IsServer);
+        }
+        if (Ctx->CloseCount == Ctx->CurrentCount++) {
+            Conn->Close();
+        }
+
+        return Status;
```

Comment:
Just remove the `Status` variable and always return `QUIC_STATUS_SUCCESS` here instead.